client/scripts/views/modals/create_community_modal.ts

```diff
@@ -134,33 +138,195 @@ const CreateCommunityModal: m.Component<IAttrs, IState> = {
+          }
+        },
+      }),
+      ]),
+    ]);
+  }
+};
+
+interface SubstrateFormAttrs {}
+
+interface SubstrateFormState {
+  name: string,
+  nodeUrl: string,
+  symbol: string,
+  website: string,
+  discord: string,
+  element: string,
+  telegram: string,
+  github: string,
+  description: string,
+}
+
+const SubstrateForm: m.Component<SubstrateFormAttrs, SubstrateFormState> = {
+  oninit: (vnode) => {
+    vnode.state.name = '';
+    vnode.state.nodeUrl = '';
+    vnode.state.symbol = '';
+    vnode.state.website = '';
+    vnode.state.discord = '';
+    vnode.state.element = '';
+    vnode.state.telegram = '';
+    vnode.state.github = '';
+    vnode.state.description = '';
+  },
+  view: (vnode) => {
+    return m('.compact-modal-body-max', [
+      m('.CommunityMetadataManagementTable', [m(Table, {
+        bordered: false,
+        interactive: false,
+        striped: false,
+        class: 'metadata-management-table',
+      }, [
+        m(InputPropertyRow, {
+          title: 'Name',
+          defaultValue: vnode.state.name,
+          onChangeHandler: (v) => { vnode.state.name = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Node URL',
+          defaultValue: vnode.state.nodeUrl,
+          placeholder: 'wss://',
+          onChangeHandler: (v) => { vnode.state.nodeUrl = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Symbol',
+          defaultValue: vnode.state.symbol,
+          placeholder: 'XYZ',
+          onChangeHandler: (v) => { vnode.state.symbol = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Description',
+          defaultValue: vnode.state.description,
+          onChangeHandler: (v) => { vnode.state.description = v; },
+          textarea: true,
+        }),
+        m(InputPropertyRow, {
+          title: 'Website',
+          defaultValue: vnode.state.website,
+          placeholder: 'https://example.com',
+          onChangeHandler: (v) => { vnode.state.website = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Discord',
+          defaultValue: vnode.state.discord,
+          placeholder: 'https://discord.com/invite',
+          onChangeHandler: (v) => { vnode.state.discord = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Element',
+          defaultValue: vnode.state.element,
+          placeholder: 'https://matrix.to/#',
+          onChangeHandler: (v) => { vnode.state.element = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Telegram',
+          defaultValue: vnode.state.telegram,
+          placeholder: 'https://t.me',
+          onChangeHandler: (v) => { vnode.state.telegram = v; },
+        }),
+        m(InputPropertyRow, {
+          title: 'Github',
+          defaultValue: vnode.state.github,
+          placeholder: 'https://github.com',
+          onChangeHandler: (v) => { vnode.state.github = v; },
+        }),
```

Comment:
We're going to have to add another field, probably a larger one, for what's known as a "NodeSpec". This is a json blob that we have to validate (we are already doing so in the manage community modal where the admins can edit+update the NodeSpec, you can copy the validation schema from there). This "nodespec" is added to the `chain.ts` model on the server.